hail/src/main/scala/is/hail/expr/types/physical/PInterval.scala

```diff
@@ -88,44 +81,35 @@ case class PInterval(pointType: PType, override val required: Boolean = false) e
       }
     }
 
-  val representation: PStruct = PStruct(
```

Comment:
based on your comment, it sounded like type signature shouldn't live here. As noted above, we should either have the type signatures on the abstract classes, set ComplexPType's representation as an instance of PStruct (which may have other side effects, haven't checked), change implementation to make representation private (and remove its use by callers), or have callers match only on the concrete class when it needs representation.

Reply:
I think it *shouldn't* live here, but we need more progress on infrastructure to make that happen. The right first step is to do this refactor, with the abstract interfaces (e.g. PInterval) still extending ComplexPType, so that their users can still use e.g. representation.

Reply:
You can do what you've done in the other PR: `override lazy val representation: PStruct = structRepresentation` and `def structRepresentation: PStruct`

Reply:
Ok, I will issue a separate PR, once I figure out the PStruct bug

Reply:
https://github.com/hail-is/hail/issues/7746